src/utils.rs

```diff
@@ -214,6 +215,23 @@ pub fn parse_tag(input: Option<&str>) -> Option<String> {
     }
 }
 
+pub fn get_uuid_from_url(url: &str) -> String {
+    if let Ok(url) = Url::parse(url) {
+        let path_segments: Vec<_> = url.path_segments().unwrap().collect();
+        if let Some(uuid_segment) = path_segments.get(3) {
+            if uuid_segment.len() == 36 {
+                uuid_segment.to_string()
+            } else {
+                "".to_string()
+            }
+        } else {
+            "".to_string()
+        }
+    } else {
+        "".to_string()
+    }
+}
+
```

Comment:
```suggestion
pub fn get_uuid_from_url(url: &str) -> Option<&str> {
    let url = Url::parse(url).ok()?;
    let path_segments: Vec<_> = url.path_segments()?.collect();
    let uuid_segment = path_segments.get(3)?;
    if uuid_segment.len() == 36 {
        Some(uuid_segment.to_string())
    } else {
        None
    }
}

```

Reply:
Nice! Thanks Adam, good to hear from you! Hope the new gig is going great!

I will surely do this. I went down this path at first, but couldn't reconcile the type I was sending into the caller (and Option return type) and got it to the point where "it works" in hopes that the review would shake out the fumbling-my-way-through-it approach! ;)

Also, I still feel it's not quite a useful utility function in that it's quite specific to the URL pattern "http://<host>/<api_path>/<api_version>/<object>/<uuid>/" where the UUID has to be in the 3 position, but I also don't care to overcomplicate it, so moving the function into the projects file seems prudent, especially since that's all I intended for this "feature"?